include/selene/primitives.h

```diff
@@ -94,11 +94,17 @@ inline T _check_get(_id<T&&>, lua_State *l, const int index) {
 
 
 inline int _check_get(_id<int>, lua_State *l, const int index) {
+#if LUA_VERSION_NUM >= 503
+    return luaL_checkinteger(l, index);
+#else
     return luaL_checkint(l, index);
```

Comment:
should just use `luaL_checkinteger` in all cases. `luaL_checkint` is only a cast.